Examples/Examples/All Examples/SwiftUIExample.swift

```diff
@@ -0,0 +1,114 @@
+import UIKit
+import MapboxMaps
+import Turf
+import SwiftUI
+
+/// Example of how a SwiftUI map view can be used within a typical `ContentView`.
+public struct ContentView: View {
+
+    public var body: some View {
+        SwiftUIMapView(resourceOptions: ResourceOptions(accessToken: AccountManager.shared.accessToken!))
+            .styleURL(.dark)
+            .zoomLevel(14.0)
+            .centerCoordinate(CLLocationCoordinate2D(latitude: 20.6905, longitude: -88.2024))
+            .on(.mapLoadingFinished) { ( mapView ) in
+                let pointAnnotation = PointAnnotation(coordinate: mapView.centerCoordinate)
+                mapView.annotationManager.addAnnotation(pointAnnotation)
+            }
+    }
+}
+
+/// To render a `MapView` as a SwiftUI view, configure a new
+/// struct that implements the required `UIViewRepresentable` protocol.
+public struct SwiftUIMapView: UIViewRepresentable {
+
+    let resourceOptions: ResourceOptions
+    internal let observerConcrete: ObserverConcrete // Manages the observation of map events
+    internal var mapView: MapView
+
+    init(resourceOptions: ResourceOptions) {
+        self.resourceOptions = resourceOptions
+        self.mapView = MapView(with: .zero, resourceOptions: resourceOptions)
+        self.observerConcrete = ObserverConcrete(mapView: self.mapView)
+
+        // Subscribe to map events
+        let events = MapEvents.EventKind.allCases.map({ $0.rawValue })
+        try? mapView.__map.subscribe(for: observerConcrete, events: events)
+    }
+
+    // MARK: - UIViewRepresentable required methods
+    public func makeUIView(context: UIViewRepresentableContext<SwiftUIMapView>) -> MapboxMaps.MapView {
+        return mapView
+    }
+
+    public func updateUIView(_ uiView: MapView, context: Context) {
+        // Unimplemented
+    }
+
+    // MARK: - Setting basic map properties
+    public func styleURL(_ newValue: StyleURL) -> SwiftUIMapView {
+        mapView.style.styleURL = newValue
+        return self
+    }
+
+    public func zoomLevel(_ newValue: CGFloat) -> SwiftUIMapView {
+        mapView.cameraManager.setCamera(zoom: newValue)
+        return self
+    }
+
+    public func centerCoordinate(_ newValue: CLLocationCoordinate2D) -> SwiftUIMapView {
+        mapView.cameraManager.setCamera(centerCoordinate: newValue)
+        return self
+    }
+
+    // MARK: - Responding to events
+    // Allows the SwiftUI map view to respond to map events coming from the internal `MapView`.
+    public func on(_ eventType: MapEvents.EventKind, handler: @escaping (MapboxMaps.MapView) -> Void) -> SwiftUIMapView {
+        var handlers: [(MapboxMaps.MapView) -> Void] = observerConcrete.eventHandlers[eventType.rawValue] ?? []
+        handlers.append(handler)
+        observerConcrete.eventHandlers[eventType.rawValue] = handlers
+        return self
```

Comment:
Is it possible to pass through to `MapView`'s `on`? And so do away with the need for `ObserverConcrete`?

Reply:
My first attempt was to try to have `SwiftUIMapView` conform to `Observer` to prevent having to make this class, but it turns out that's a class-bound protocol which prevented me from doing that. We should eventually investigate the feasibility of making a change to core to allow for `AnyObject` to conform to `Observer`.

In addition, the `.on(_:handler)` method is on `BaseMapView`, and the `SwiftUIMapView` is a struct which limits our ability to subclass from it.